dm/master/server.go

```diff
@@ -450,23 +447,27 @@ func (s *Server) StartTask(ctx context.Context, req *pb.StartTaskRequest) (*pb.S
 		for _, stCfg := range stCfgs {
 			sources = append(sources, stCfg.SourceID)
 		}
-		s.removeMetaLock.Lock()
 		if req.RemoveMeta {
+			release, err3 := s.scheduler.AcquireSubtaskLatch(cfg.Name)
```

Comment:
Will there be a situation where：

Two start-task requests A and B, A requires remove-meta but B does not

t1: A grabs the task's lock (starts remove-meta)
t2: A finishes the remove-meta and releases the lock on the task, but at the same time request B is called in
t2: B grabs the lock on the task and starts execution, but A returns with an exception

Is this not as expected? Perhaps adding a lock to the start of each start-task request would solve this problem?

Reply:
the result is same, meta removed and task started

Reply:
but if request A task.yaml is different B task.yaml (only task name is same)?

I know this may rarely happen 😂, But I think we should be more critical?

Reply:
we didn't provide transactional guarantee of DM API (for example, some subtask will fail while other succeeded. so remove meta may success and start task failed) 🤔 can you propose a boundary of API guarantee

Reply:
> the result is same, meta removed and task started

Think it over. The result is not really the same
Since A reaches the server first, request A should be fully executed before request B is executed

The expected situation is that when request B reaches the server, it should either return that task is being created, or that task has already been created

Reply:
There is no requirement that if a task is created all subtasks are created successfully

But each individual request should not be interrupted by concurrent requests
In the previous implementation, when a remote-meta was required, each start-task would
1. get the remove-meta lock 
2. get master-lock
3. release master-lock
4. release the remove-meta lock

With this implementation, each request is independent and atomic